### DISCUSSION TITLE
tf_2dunet Erorr

### DISCUSSION THREAD

**Discussion by omnyaatef22** [securefederatedai/openfederatedlearning#366] [Q&A]
![Capture](https://user-images.githubusercontent.com/98268106/156519762-74d24fc0-752d-499b-b402-31c2e318bcd2.PNG)
trying to initialize the plan but I can't

**Comment by itrushkin** [comment]
Hi @omnyaatef22. Thank you for your interest in OpenFL!

Because of legal restriction of [BraTS dataset](https://www.med.upenn.edu/cbica/brats2019/data.html), `tf_2dunet` workspace requires data to be already on the device. If you have the data locally, could you provide directory tree of the dataset? 
You can do it via executing:
```bash
sudo apt-get install tree
tree -L 2 path/to/data/MICCAI_BraTS_2019_Data_Training/HGG # replace with your location
```
UPD: The code above works on Ubuntu system, but you can try using any Windows alternative you prefer.

**Reply by omnyaatef22** [reply]
![1](https://user-images.githubusercontent.com/98268106/160230607-7134eb3d-4722-42a4-9fdd-5a6ea06b1c1d.PNG)
![2](https://user-images.githubusercontent.com/98268106/160230613-0210f6de-9815-4e60-b1eb-16df6698303f.PNG)
![3](https://user-images.githubusercontent.com/98268106/160230620-6b13c24b-0c02-4787-a271-7e7d79aec44f.PNG)
![4](https://user-images.githubusercontent.com/98268106/160230660-5f53a1a9-63cc-46f5-ab87-c0fba44bea41.PNG)
![5](https://user-images.githubusercontent.com/98268106/160230663-aa2e0949-78fe-44fc-83f2-91ba3ae4c8b3.PNG)

**Reply by omnyaatef22** [reply]
![0000](https://user-images.githubusercontent.com/98268106/160231318-e4dad776-d7b5-4a57-893c-4bd44543aa7f.PNG)

i think the exception is here

**Comment by itrushkin** [comment]
@omnyaatef22 Your directory tree looks correct. The error tells that there are no sub-folders in the specified `data_path`. Path listed in `tree` command is not the path specified in the error. This could be a mistake.

**Reply by omnyaatef22** [reply]
There are no mistakes. It's just that the screenshots were taken from 2 different attempts with two copies of the dataset in different folders in each attempt. Here are screenshots from the same attempt with the same error as before.
![1](https://user-images.githubusercontent.com/98268106/161375868-13fcf8c9-e07f-4cb5-9e0e-ea83333a0fde.PNG)
![2](https://user-images.githubusercontent.com/98268106/161375871-735977d8-6e3f-4b3c-ae4f-11e410ec27df.PNG)
![3](https://user-images.githubusercontent.com/98268106/161375873-06a17745-faf3-4610-a438-aa4cb40cdf9c.PNG)
![4](https://user-images.githubusercontent.com/98268106/161375877-87a8b37a-8897-4d5a-8df7-35b0240c3087.PNG)
![5](https://user-images.githubusercontent.com/98268106/161375879-138ca5fa-4033-46c5-9274-d30b90f6d49e.PNG)
![6](https://user-images.githubusercontent.com/98268106/161375880-e7ffe875-14ce-4dda-8dfa-52063ee9dae0.PNG)

**Reply by itrushkin** [reply]
@omnyaatef22 However, the error indicates that there are no images in the specified directory or there was an error during reading them.
Could you please execute the command in debug mode for more details?
``` bash
fx -l DEBUG plan initialize
```
and attach the resulting output.
Also, just in case, could you share a deeper directory tree?

**Reply by omnyaatef22** [reply]
![1](https://user-images.githubusercontent.com/98268106/162171937-556cfacf-55d0-4922-b35c-8c3166aa68bf.PNG)
![2](https://user-images.githubusercontent.com/98268106/162171943-67dd740a-2740-47a8-8c
![3](https://user-images.githubusercontent.com/98268106/162171953-bd5499a1-c3ab-485b-8263-f7adf11625a8.PNG)
49-cd53ee9e0cbf.PNG)
![4](https://user-images.githubusercontent.com/98268106/162171967-e555cf31-b4b9-4a86-8144-a9734cf41de5.PNG)
![5](https://user-images.githubusercontent.com/98268106/162171998-d3694c93-fddc-470b-b0c7-3f7616b0346b.PNG)
![6](https://user-images.githubusercontent.com/98268106/162172152-a2ea0825-0405-4768-acfc-1a3357dae996.PNG)

**Reply by omnyaatef22** [reply]
template file 
![7](https://user-images.githubusercontent.com/98268106/162173304-ecaeb48d-3d7b-4259-84d4-0c56293b2fb2.PNG)
![8](https://user-images.githubusercontent.com/98268106/162173341-a0070771-bc49-4e0a-970a-678caecc1c30.PNG)

**Comment by omnyaatef22** [comment]
![9](https://user-images.githubusercontent.com/98268106/162187817-d9234e92-2fc7-463b-8860-b9590a219481.PNG)
![10](https://user-images.githubusercontent.com/98268106/162187821-8f7dbb78-b29e-4306-a31e-73bdeaeb0f81.PNG)
